packages/lesswrong/lib/collections/conversations/schema.ts

```diff
@@ -116,7 +115,54 @@ const schema: SchemaType<"Conversations"> = {
       );
       const filteredMessage = await accessFilterSingle(currentUser, context.Messages, message, context)
       return filteredMessage;
-    }
+    },
+    sqlResolver: ({field, join}) => join({
+      table: "Messages",
+      type: "left",
+      on: {
+        _id: `(
+          SELECT "_id"
+          FROM "Messages"
+          WHERE "conversationId" = ${field("_id")}
+          ORDER BY "createdAt" DESC
+          LIMIT 1
+        )`,
+      },
+      resolver: (messagesField) => messagesField("*"),
+    }),
+  }),
+  hasUnreadMessages: resolverOnlyField({
+    type: Boolean,
+    graphQLtype: "Boolean",
+    canRead: ["members"],
+    resolver: async (
+      conversation: DbConversation,
+      _args,
+      context: ResolverContext,
+    ) => {
+      const {currentUser} = context;
+      if (!currentUser) {
+        return false;
+      }
+      return getWithCustomLoader<boolean, string>(
+        context,
+        "hasUnreadMessages",
+        conversation._id,
+        (conversationIds: string[]): Promise<boolean[]> =>
+          context.repos.conversations.getReadStatuses(currentUser._id, conversationIds),
+      );
+    },
+    sqlResolver: ({field, currentUserField, join}) => join({
+      isNonCollectionJoin: true,
```

Comment:
This `isNonCollectionJoin` is a new flag to allow us to join on things that aren't collections (in this case, it's a view).

Reply:
Nice